src/utilities/resolveRoutes.ts

```diff
@@ -32,11 +34,16 @@ export function resolveRoutes(routes: Routes, parentContext: ParentContext = {})
         matched: markRaw(route),
         matches: markRaw(fullMatches),
         name: route.name,
-        path: fullPath,
-        params: fullParams,
+        path: fullPath.map(({ path }) => path.toString()).join(''),
+        query: fullQuery.map(({ query }) => query.toString()).join('&'),
+        params: mergeParams(reduceParams(fullPath), reduceParams(fullQuery)),
       })
     }
 
     return value
   }, [])
+}
+
+function reduceParams(entries: { params: Record<PropertyKey, unknown> }[]): Record<PropertyKey, Param[]> {
```

Comment:
Does this not work?
```suggestion
function reduceParams(entries: Path[] | Query[]): Record<string, Param[]> {
```

Curious about why the generic record type for the params and also why `PropertyKey` rather than `string` for the return type.

Reply:
yep, both seem fine 👍